internal/indexmanagement/reconcile.go

```diff
@@ -140,65 +134,47 @@ func ReconcileCurationConfigmap(apiclient client.Client, cluster *apis.Elasticse
+	containers := []*corev1.Container{}
+	if policy.Phases.Delete != nil {
 		return nil
-	}
-	schedule, err := crontabScheduleFor(policy.PollInterval)
-	if err != nil {
-		return kverrors.Wrap(err, "failed to reconcile rollover cronjob", "policymapping", mapping.Name)
-	}
-	conditions := calculateConditions(policy, primaryShards)
-	name := fmt.Sprintf("%s-rollover-%s", cluster.Name, mapping.Name)
-	payload, err := json.Marshal(map[string]rolloverConditions{"conditions": conditions})
-	if err != nil {
-		return kverrors.Wrap(err, "failed to serialize the rollover conditions to JSON")
-	}
-	envvars := []corev1.EnvVar{
-		{Name: "PAYLOAD", Value: base64.StdEncoding.EncodeToString(payload)},
-		{Name: "POLICY_MAPPING", Value: mapping.Name},
-	}
-	fnContainerHandler := func(container *corev1.Container) {
-		container.Command = []string{"bash"}
-		container.Args = []string{
-			"-c",
-			"/tmp/scripts/rollover",
+		minAgeMillis, err := calculateMillisForTimeUnit(policy.Phases.Delete.MinAge)
+		if err != nil {
+			return err
 		}
+		envvars := []corev1.EnvVar{
+			{Name: "POLICY_MAPPING", Value: mapping.Name},
+			{Name: "MIN_AGE", Value: strconv.FormatUint(minAgeMillis, 10)},
+		}
+		containers = append(containers, newInitContainer(cluster.Name, "delete", constants.PackagedElasticsearchImage(), "/tmp/scripts/delete", envvars))
+	} else {
+		log.V(1).Info("Skipping curation cronjob for policymapping; delete phase not defined", "policymapping", mapping.Name)
 	}
-	desired := newCronJob(cluster.Name, constants.PackagedElasticsearchImage(), cluster.Namespace, name, schedule, cluster.Spec.Spec.NodeSelector, cluster.Spec.Spec.Tolerations, envvars, fnContainerHandler)
 
-	cluster.AddOwnerRefTo(desired)
-	return reconcileCronJob(apiclient, cluster, desired, areCronJobsSame)
-}
-
-func ReconcileCurationCronjob(apiclient client.Client, cluster *apis.Elasticsearch, policy apis.IndexManagementPolicySpec, mapping apis.IndexManagementPolicyMappingSpec, primaryShards int32) error {
-	if policy.Phases.Delete == nil {
-		log.Info("Skipping curation cronjob for policymapping; delete phase not defined", "policymapping", mapping.Name)
-		return nil
+	if policy.Phases.Hot != nil {
+		conditions := calculateConditions(policy, primaryShards)
+		payload, err := json.Marshal(map[string]rolloverConditions{"conditions": conditions})
+		if err != nil {
+			return kverrors.Wrap(err, "failed to serialize the rollover conditions to JSON")
+		}
+		envvars := []corev1.EnvVar{
+			{Name: "PAYLOAD", Value: base64.StdEncoding.EncodeToString(payload)},
+			{Name: "POLICY_MAPPING", Value: mapping.Name},
+		}
+		containers = append(containers, newInitContainer(cluster.Name, "rollover", constants.PackagedElasticsearchImage(), "/tmp/scripts/rollover", envvars))
+	} else {
+		log.V(1).Info("Skipping rollover cronjob for policymapping; hot phase not defined", "policymapping", mapping.Name)
```

Comment:
similar as above

Reply:
`log.V(1).Info("Skipping rollover management for policymapping; hot phase not defined", "policymapping", mapping.Name)`